webapp/lib/controller.dart

```diff
@@ -284,33 +233,52 @@ void command(UIAction action, Data actionData) {
       break;
     case UIAction.addConfigurationResponseEntries:
       ConfigurationResponseData data = actionData;
-      addConfigurationResponseEntries(data.parentTag);
+      addConfigurationResponseEntries(data.parentTag, data.language, data.text);
       break;
   }
 }
 
-void showConfigurationView() {
-  view.contentView.configurationView.tagList.renderTagList(tagData.keys.toList());
-  view.contentView.configurationView.tagResponses.renderResponses(tagData.keys.toList().first, tagData.values.toList().first);
+void showConfigurationView([String selectedTag, Map<String, Map<String, List<String>>> filteredTagData]) {
+  Map<String, bool> tags = new Map.fromIterable(tagData.keys.toList(),
+    key: (tag) => tag,
+    value: (tag) => selectedTag != null && selectedTag == tag ? true : false);
+  if (selectedTag == null) tags[tags.keys.toList()[0]] = true;
+  view.contentView.configurationView.tagList.renderTagList(tags);
+  if (filteredTagData != null) {
+    view.contentView.configurationView.tagResponses.renderResponses(filteredTagData.keys.toList().first, filteredTagData.values.toList().first);
+  } else {
+    view.contentView.configurationView.tagResponses.renderResponses(tagData.keys.toList().first, tagData.values.toList().first);
+  }
   view.contentView.renderView(view.contentView.configurationView.configurationViewElement);
 }
 
 void retrieveConfigurationTagResponse(String selectedTag) {
-  selectedConfigurationTag = selectedTag;
-  var filteredTagResponses = Map<String, Map<String, List<String>>>.from(tagData)..removeWhere((k, v) => !k.contains(selectedTag));
-  view.contentView.configurationView.tagResponses.renderResponses(filteredTagResponses.keys.toList().first, filteredTagResponses.values.toList().first);
+  var filteredTagData = Map<String, Map<String, List<String>>>.from(tagData)..removeWhere((k, v) => !k.contains(selectedTag));
```

Comment:
`filtered` is not a very good name here - what you're doing is taking the messages corresponding to the selected tag. how about `messagesForTag`?
Also, wouldn't it be more straightforward to get this data by doing:
```suggestion
  var filteredTagData = tagData[selectedTag];
```
(which you also don't really need to wrap in a method)

Reply:
Done some data refactoring as per your previous comment. I've used `selectedTagResponses`